README.md

```diff
@@ -1,145 +1,231 @@
-# babel-plugin-transform-react-pug
+# babel-plugin-transform-react-pug [![npm version](https://img.shields.io/npm/v/babel-plugin-transform-react-pug.svg)](https://www.npmjs.com/package/babel-plugin-transform-react-pug)
 
-The official means by which you can use pug in your react components, replaces the use of react-jade when moving to pug.
+Use Pug templates to write react components.
 
-This plugin transforms the pug inside of your react components.
+It's a plugin for babel which transpiles pug syntax within template literals to jsx.
 
-## Installation
+Write your components this way:
+
+```jsx
+export default const ReactComponent = props => pug`
+  .wrapper
+    if props.shouldShowGreeting
+      p.greeting Hello World!
+
+    button(onClick=props.notify) Click Me
+`
 ```
-npm install babel-plugin-transform-react-pug --save-dev
-npm install babel-plugin-transform-react-jsx --save-dev
+
+And it will be transpiled into:
+
+```jsx
+export default const ReactComponent = props => (
+  <div className="wrapper">
+    {props.shouldShowGreeting && (
+      <p className="greeting">Hello World</p>
+    )}
+    <button onClick={props.notify}>Click Me</button>
+  </div>
+)
 ```
+
+* [Usage](#usage)
+  * [Syntax](#syntax)
+  * [Eslint integration](#eslint-integration)
+  * [CSS Modules](#css-modules)
+* [Install](#install)
+  * [create-react-app](#create-react-app)
+  * [React Native](#react-native)
+* [How it works](#how-it-works)
+* [Limitations](#limitations)
+* [FAQ](#faq)
+  * [Can I import template from other files?](#can-i-import-template-from-other-files)
+  * [How to get syntax highlighting in IDE (or text editors)?](#how-to-get-syntax-highlighting-in-ide-or-text-editors)
+* [License](#license)
```

Comment:
This table of contents will be a bit of a pain to keep up to date, I'm not sure if the README is really long enough to need it.

Reply:
@ForbesLindesay I'm going to keep it just to let people understand how ready it's for production. I mean I would rather keep outdated table of contents than remove it.

Do you think we should avoid it or think up something which will generate TOC automatically? Like we can create a bot (I believe probot has something for this) which will make pushes to master branch after changing headings in README?